### ISSUE TITLE
[3.6.rc1] Existing meshes don't use vertex cache optimization?

### ISSUE THREAD

**Issue by Dadaskis** [godotengine/godot#96369]
### Tested versions

- Reproducible in 3.6.rc1.

### System information

Windows 11 - Godot v3.6.rc1 - GLES3 - dedicated

### Issue description

![image](https://github.com/user-attachments/assets/3914a99e-8ed2-4fc5-b154-7e763c877c94)

In short, i have a ton of meshes saved as .res files. That helps to save scene file sizes, helps to manipulate materials of the surfaces, etc. However, as far as i understand, vertex cache optimization affects only reimported meshes (which i did btw, i removed .import and reimported all assets). In this case, meshes saved in .res files may stay unaffected by this change. There are no methods in `Mesh` nor `ArrayMesh` to optimize the vertex cache with GDScript. So, the question is, how can i apply this optimization method on existing mesh resources? Or is it applied automatically and is not documented to be that way?

### Steps to reproduce

- Have a project that you have developed in 3.5.3.
- Save mesh resources in any format you'd wish (.res in my case)
- Convert to 3.6.rc1.
- Remove .import folder, painfully wait for it to import back because you should not have made that much assets.
- Play the game and notice no difference in framerates in any scenes.
... OR ...
- Open MRP with 3.6.rc1, it was made with version 3.5.3.

### Minimal reproduction project (MRP)

[MeshOptimizationMRP.zip](https://github.com/user-attachments/files/16823835/MeshOptimizationMRP.zip)

**Comment by AThousandShips** [comment]
To clarify here what should be discussed here is how to document this, for support questions please ask in the [other community channels](https://godotengine.org/community/), not here

**Comment by Dadaskis** [comment]
> To clarify here what should be discussed here is how to document this, for support questions please ask in the [other community channels](https://godotengine.org/community/), not here

Excuse me if i didn't say it correctly, but the problem here is also related to resource files. Meshes that were saved in .res files seem to have no vertex cache optimization.

**Comment by AThousandShips** [comment]
Yes and that's what needs to be discussed and documented if the current behavior is expected, but the question of "how can i apply this optimization method on existing mesh resources?" is a support question and not a bug report, at least in the way it's phrased

**Comment by Dadaskis** [comment]
> that's what needs to be discussed and documented if the current behavior is expected

I didn't expect that behavior when i switched to 3.6.rc1 to test how would vertex cache optimize my game and it is not documented anywhere as far as i know, please, correct me if i'm wrong, but i really could not find any information, hence why i thought it would be worthy to report because, in my opinion, Lawnjelly may need about this

**Comment by lawnjelly** [comment]
Have just seen this, it is fine to report as an issue imo.

The situations with already imported resources didn't occur to me. I'm not sure there is an easy way around this, as we don't want to be applying it to already imported resources "just in case".

Perhaps it is possible to write some kind of upgrade script, I'll have a think.

_Bear in mind also this is a "nice to have" optimization, there are no promises for a dramatic difference, depending on your game and asset type (it gives more benefit with higher poly models, which are more likely to be bottlenecked by vertex processing)._
